### ISSUE TITLE
devtools signal lookup incorrectly displays signal with empty string

### ISSUE THREAD

**Issue by ducin** [angular/angular#55727]
### Which @angular/* package(s) are the source of the bug?

Devtools

### Is this a regression?

Yes

### Description

<img width="373" alt="image" src="https://github.com/angular/angular/assets/375027/9d9455e9-cf2b-47e5-997f-ececbcbc5e32">

### Please provide a minimal reproduction of the bug

just put this on a component:
`value = signal('')`

### Please provide the environment you discovered this bug in (run `ng version`)

```true
`@angular/core@17.3.4`
```

**Comment by ducin** [comment]
@JeanMeche @AleksanderBodurri  We could either do a dirty ugly fix (yet another `if` in a bad place) or do a proper refactor. I prefer to do the latter. **I'm halfway of fixing this already**, hope to get it done by the end of the week.

**Comment by AleksanderBodurri** [comment]
Agreed that we can go with a proper refactor here. Thank you for reporting this @ducin 🙏  Let me know if you need some assistance 😄
